test/e2e_appium/tests/test_onboarding_flow.py

```diff
@@ -49,15 +49,41 @@ def test_onboarding_new_password_skip_analytics(self, onboarded_user):
 
         for step in expected_steps:
             assert step in completed_steps, f"Step '{step}' should be completed"
-        # Validate analytics action matches config
+
         assert result["step_results"]["analytics_screen"]["action"] == "skipped"
 
         self.logger.info("Complete onboarding flow test with fixture passed!")
 
     @pytest.mark.onboarding
     @lambdatest_reporting
-    @pytest.mark.onboarding_config(custom_display_name="E2E_TestUser")
-    def test_onboarding_lands_on_main_app(self, onboarded_app):
+    @pytest.mark.onboarding_config(
+        custom_display_name="E2E_TestUser",
+        validate_each_step=True,
+        create_profile_method="password",
+    )
+    def test_onboarding_create_with_password_lands_on_main_app(self, onboarded_app):
         app = onboarded_app
         assert app.is_main_app_loaded()
         assert app.user_data["display_name"] == "E2E_TestUser"
+
+    @pytest.mark.onboarding
+    @lambdatest_reporting
+    @pytest.mark.onboarding_config(
+        custom_display_name="E2E_TestUser",
+        validate_each_step=True,
+        create_profile_method="seed_phrase",
+    )
+    def test_onboarding_create_with_seed_lands_on_main_app(self, onboarded_app):
+        app = onboarded_app
+        assert app.is_main_app_loaded()
+
+    @pytest.mark.onboarding
+    @lambdatest_reporting
+    @pytest.mark.onboarding_config(
+        custom_display_name="E2E_TestUser",
+        validate_each_step=True,
+        create_profile_method="random",
+    )
+    def test_onboarding_create_random_lands_on_main_app(self, onboarded_app):
```

Comment:
i am little bit confused it is delivered as a single test, that should be a precondition for any test to start with:
- you either create a new user with password
- or you create new profile with seed
- or you login to the app that has already a data folder (aka Returning user)

in all these cases the end result is landing to main screen

suggest to redo it the way it could be one fixture with variety of choices (new user, or returning user) and pass it to test